java/src/jmri/jmrix/secsi/SerialSensorManager.java

```diff
@@ -52,10 +54,14 @@ public void dispose() {
     }
 
     /**
-     * Create a new sensor if all checks are passed. System name is normalized to
-     * ensure uniqueness.
+     * {@inheritDoc}
+     * <p>
+     * System name is normalized to ensure uniqueness.
+     *
+     * @return null if the system name is not in a valid format (TODO change that to throw an exception, Spotbugs)
```

Comment:
The ``createNewSensor`` abstract method is annotated as ``@Nonnull`` and the javadoc states ``@return never null``. But many jmrix implementations do return ``null`` if eg. an invalid name was provided. That makes Spotbugs flagging a warning for overriding the base annotation.
Is it true that this should somehow be improved, requiring name validation before calling this method?
(Same applies to ``createNewTurnout``, ``createNewMemory`` etc.)

Reply:
If an invalid name is provided, it's better to throw an `IllegalArgumentException` than returning `null`.

Reply:
Moved this to a new Issue #7661